src/Elmish.WPF/Binding.fs

```diff
@@ -221,11 +221,14 @@ module Binding =
     let private mapMinorTypes
         (outMapBindingModel: 'bindingModel -> 'bindingModel0)
         (outMapBindingMsg: 'bindingMsg -> 'bindingMsg0)
+        (outMapBindingViewModel: 'bindingViewModel -> 'bindingViewModel0)
         (inMapBindingModel: 'bindingModel0 -> 'bindingModel)
         (inMapBindingMsg: 'bindingMsg0 -> 'bindingMsg)
-        (d: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg>) = {
+        (inMapBindingViewModel: 'bindingViewModel0 -> 'bindingViewModel)
+        (d: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>) = {
       GetModel = d.GetModel >> ValueOption.map outMapBindingModel
-      GetBindings = d.GetBindings >> Bindings.mapModel inMapBindingModel >> Bindings.mapMsg outMapBindingMsg
+      CreateViewModel = fun args -> d.CreateViewModel(args |> ViewModelArgs.map inMapBindingModel outMapBindingMsg) |> outMapBindingViewModel
```

Comment:
This is ultimately what allows us to have properly typed `ViewModel<'bindingModel,'bindingMsg>`'s in our collections that are only boxed once back to `obj` instead of always created as `ViewModel<obj,obj>` with boxed/unboxed inputs/outputs.